services/motion/builtin/builtin.go

```diff
@@ -122,18 +127,31 @@ func (ms *builtIn) Reconfigure(
 	ms.movementSensors = movementSensors
 	ms.slamServices = slamServices
 	ms.components = components
+	if ms.state != nil {
```

Comment:
When would this be nil?

Reply:
when it is called from the constructor: https://github.com/viamrobotics/rdk/blob/a4abb90482235b692a061d13660bc804a6f17a9d/services/motion/builtin/builtin.go#L86

Reply:
Ah I see whats happening I think I read this PR in the wrong order.